### ISSUE TITLE
[java] MissingStaticMethodInNonInstantiatableClass for constructor with lombok.Builder annotation

### ISSUE THREAD

**Issue by kdebski85** [pmd/pmd#5108]
**Affects PMD Version:** 
7.3.0

**Rule:**

[MissingStaticMethodInNonInstantiatableClass](https://docs.pmd-code.org/pmd-doc-7.3.0/pmd_rules_java_errorprone.html#missingstaticmethodinnoninstantiatableclass)

**Description:**

**Code Sample demonstrating the issue:**

```java
public class LombokBuilderTest {

    @lombok.Builder
    private LombokBuilderTest() {

    }
}
```

**Expected outcome:**

PMD reports a violation at line 1, but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*

**Comment by jsotuyod** [comment]
Duplicate of #1488
